### PR TITLE
Added to Cart Toast Component

### LINKED ISSUE #13346: Added to Cart Toast Component
View that shows up after adding a product to cart.

### PR DESCRIPTION
**Pull request by bit172**
close #13346

### CODE REVIEW

File: feature-libs/cart/added-to-cart-toast/components/added-to-cart-toast/added-to-cart-toast.component.html [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      </div>
+      <span class="continue-shopping" (click)="closeToast(toast)">
+        Continue Shopping
+      </span>
+      <a
+        [routerLink]="{ cxRoute: 'cart' } | cxUrl"
+        (click)="closeToast(toast)"
+        class="btn btn-primary btn-block"
+      >
+        Go to Cart
```
Comment (dydome) [inline feature-libs/cart/added-to-cart-toast/components/added-to-cart-toast/added-to-cart-toast.component.html]: missing translation

File: feature-libs/cart/added-to-cart-toast/components/added-to-cart-toast/added-to-cart-toast.component.html [hunk 2]
```diff
@@ -0,0 +1,30 @@
+<ng-container *ngIf="cartToasts$ | async as cartToasts">
+  <div *ngFor="let toast of cartToasts" [class]="toast.baseClass">
+    <div class="added-to-cart-toast-title">
+      {{ toast.quantity }} {{ toast.quantity > 1 ? 'items' : 'item' }} added
```
Comment (dydome) [inline feature-libs/cart/added-to-cart-toast/components/added-to-cart-toast/added-to-cart-toast.component.html]: missing translation

File: feature-libs/cart/added-to-cart-toast/components/added-to-cart-toast/added-to-cart-toast.component.html [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        format="product"
+        [alt]="toast.productName"
+      ></cx-media>
+      <div class="added-to-cart-toast-product">
+        <h3>{{ toast.productName }}</h3>
+        <span>Qty: {{ toast.quantity }}</span>
+        <span>Price: {{ toast.unitPrice }}</span>
+      </div>
+      <span class="continue-shopping" (click)="closeToast(toast)">
+        Continue Shopping
```
Comment (dydome) [inline feature-libs/cart/added-to-cart-toast/components/added-to-cart-toast/added-to-cart-toast.component.html]: missing translation

File: feature-libs/cart/added-to-cart-toast/components/added-to-cart-toast/added-to-cart-toast.component.ts [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+
+@Component({
+  selector: 'cx-added-to-cart-toast',
+  templateUrl: './added-to-cart-toast.component.html',
+})
+export class AddedToCartToastComponent implements OnInit, OnDestroy {
+  protected subscription = new Subscription();
+
+  quantity: number;
+
```
Comment (dydome) [inline feature-libs/cart/added-to-cart-toast/components/added-to-cart-toast/added-to-cart-toast.component.ts:30]: why there are so many enters between each row? also I will suggest putting HostBinding below standard variables ok?

**Review by dydome — CHANGES_REQUESTED** [review]

File: feature-libs/cart/added-to-cart-toast/assets/translations/translations.ts [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import { TranslationChunksConfig, TranslationResources } from '@spartacus/core';
+import { en } from './en/index';
+
+export const addedToCartTranslations: TranslationResources = {
```
Comment (dydome) [inline feature-libs/cart/added-to-cart-toast/assets/translations/translations.ts]: Maybe use one approach with naming ok `addedToCartToastTranslation`

File: feature-libs/cart/added-to-cart-toast/components/added-to-cart-toast/added-to-cart-toast.component.ts [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+  customClass?: string;
```
Comment (dydome) [inline feature-libs/cart/added-to-cart-toast/components/added-to-cart-toast/added-to-cart-toast.component.ts:31]: I am not sure but in components, we are not using `?` in variables

File: feature-libs/cart/added-to-cart-toast/components/added-to-cart-toast/added-to-cart-toast.component.ts [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+    protected winRef: WindowRef,
+    protected addedToCartToastConfig: AddedToCartToastConfig,
+    protected addedToCartToastService: AddedToCartToastService,
+    protected eventService: EventService,
+    protected productService: ProductService,
+    protected renderer: Renderer2,
+    protected routerService: RoutingService
+  ) {}
+
+  ngOnInit() {
```
Comment (dydome) [inline feature-libs/cart/added-to-cart-toast/components/added-to-cart-toast/added-to-cart-toast.component.ts]: : void

also this ngOnInitis so huge, may you split it into another method?

File: feature-libs/cart/added-to-cart-toast/components/added-to-cart-toast/added-to-cart-toast.component.ts [hunk 8]
```diff
@@ -0,0 +1,185 @@
...
+        )
+        .subscribe(({ quantityAdded, product }) => {
+          let toastContainerClass = `${this.toastContainerBaseClass} `;
+          const toastItem = this.addedToCartToastService.addToast(
+            quantityAdded,
+            product,
+            toastContainerClass
+          );
+          // trigger enter animations with class change
+          setTimeout(() => {
```
Comment (dydome) [inline feature-libs/cart/added-to-cart-toast/components/added-to-cart-toast/added-to-cart-toast.component.ts]: what is the purpose of setTimeout without any delay? If it is related to change detector maybe you can try it in the other way?

File: feature-libs/cart/added-to-cart-toast/components/added-to-cart-toast/added-to-cart-toast.component.ts [hunk 9]
```diff
@@ -0,0 +1,185 @@
...
+
+          // trigger slide out animation after timeout
+          toastItem.timeoutRef = setTimeout(() => {
+            this._closeToast(toastItem);
+          }, this.timeout);
+        })
+    );
+  }
+
+  ngOnDestroy() {
```
Comment (dydome) [inline feature-libs/cart/added-to-cart-toast/components/added-to-cart-toast/added-to-cart-toast.component.ts]: : void

File: feature-libs/cart/added-to-cart-toast/components/added-to-cart-toast/added-to-cart-toast.component.ts [hunk 10]
```diff
@@ -0,0 +1,185 @@
...
+    this.subscription.unsubscribe();
+  }
+
+  triggerScrollEvent() {
```
Comment (dydome) [inline feature-libs/cart/added-to-cart-toast/components/added-to-cart-toast/added-to-cart-toast.component.ts]: what type this method return?

File: feature-libs/cart/added-to-cart-toast/components/added-to-cart-toast/added-to-cart-toast.component.ts [hunk 11]
```diff
@@ -0,0 +1,185 @@
...
+            quantityAdded,
+            product,
+            toastContainerClass
+          );
+          // trigger enter animations with class change
+          setTimeout(() => {
+            toastItem.baseClass = this.getToastStyles();
+          });
+
+          this.scrollEventUnlistener && this.scrollEventUnlistener();
```
Comment (dydome) [inline feature-libs/cart/added-to-cart-toast/components/added-to-cart-toast/added-to-cart-toast.component.ts]: I spoke with the architect and ask if there isany approach for short if condition in spartacus code and he said that there is no restriction to make it easier for future change mostly everybody I using standard if () {} else {} approach. May you please change it in your code?

File: feature-libs/cart/added-to-cart-toast/components/added-to-cart-toast/added-to-cart-toast.component.ts [hunk 12]
```diff
@@ -0,0 +1,185 @@
...
+  triggerScrollEvent() {
+    this.scrollEventUnlistener = this.renderer.listen(
+      this.winRef.nativeWindow,
+      'scroll',
+      this._setPosition
+    );
+  }
+
+  isHeaderInView(): boolean {
+    if (!this.headerElement) return false;
```
Comment (dydome) [inline feature-libs/cart/added-to-cart-toast/components/added-to-cart-toast/added-to-cart-toast.component.ts]: like before if () {} will be easier for changes in future

File: feature-libs/cart/added-to-cart-toast/components/added-to-cart-toast/added-to-cart-toast.component.ts [hunk 13]
```diff
@@ -0,0 +1,185 @@
...
+    const headerBounding = this.headerElement.getBoundingClientRect();
+    return headerBounding.bottom >= 0;
+  }
+
+  isHeaderFixed() {
```
Comment (dydome) [inline feature-libs/cart/added-to-cart-toast/components/added-to-cart-toast/added-to-cart-toast.component.ts:150]: return type

File: feature-libs/cart/added-to-cart-toast/components/added-to-cart-toast/added-to-cart-toast.component.ts [hunk 14]
```diff
@@ -0,0 +1,185 @@
...
+    if (!this.headerElement) return false;
+    const headerPosition = this.headerElement.style.position;
+    return headerPosition === 'fixed';
+  }
+
+  closeToast(toastItem: CartToastItem) {
```
Comment (dydome) [inline feature-libs/cart/added-to-cart-toast/components/added-to-cart-toast/added-to-cart-toast.component.ts:156]: return type

File: feature-libs/cart/added-to-cart-toast/components/added-to-cart-toast/added-to-cart-toast.component.ts [hunk 15]
```diff
@@ -0,0 +1,185 @@
...
+    toastItem.timeoutRef && clearTimeout(toastItem.timeoutRef);
+    this._closeToast(toastItem);
+  }
+
+  _closeToast = (toastItem: CartToastItem) => {
```
Comment (dydome) [inline feature-libs/cart/added-to-cart-toast/components/added-to-cart-toast/added-to-cart-toast.component.ts:163]: why _ here?

**Review by dydome — CHANGES_REQUESTED** [review]

File: feature-libs/cart/added-to-cart-toast/components/added-to-cart-toast/added-to-cart-toast.component.ts (hunk 6 above)
Comment (bit172) [reply feature-libs/cart/added-to-cart-toast/components/added-to-cart-toast/added-to-cart-toast.component.ts:31]: I was basing this component off of the `PopoverComponent` and it has a `?` there. I could remove it if it's better

File: feature-libs/cart/added-to-cart-toast/components/added-to-cart-toast/added-to-cart-toast.component.ts (hunk 8 above)
Comment (bit172) [reply feature-libs/cart/added-to-cart-toast/components/added-to-cart-toast/added-to-cart-toast.component.ts]: Yes, I'll try a different solution

File: feature-libs/cart/added-to-cart-toast/components/added-to-cart-toast/added-to-cart-toast.component.ts (hunk 15 above)
Comment (bit172) [reply feature-libs/cart/added-to-cart-toast/components/added-to-cart-toast/added-to-cart-toast.component.ts:163]: I'm trying to keep this private and it has the same name as another method. Let me know if I should change the name. I did this for `setTimeout` callbacks

**Review by dydome — APPROVED** [review]

**Pull request merged**